src/libraries/System.IO.FileSystem/tests/Directory/Delete.Windows.cs

```diff
@@ -0,0 +1,30 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Text;
+using Xunit;
+using Microsoft.DotNet.XUnitExtensions;
+using System.Security.Principal;
+using System.Security.AccessControl;
+
+namespace System.IO.Tests
+{
+    public partial class Directory_Delete_str_bool : Directory_Delete_str
+    {
+        [Fact]
+        [PlatformSpecific(TestPlatforms.Windows)]
```

Comment:
If this is only compiling on Windows does it really need the `PlatformSpecific` attribute? Perhaps you don't need to make `Directory_Delete_str_bool` partial if you keep the attribute.

Reply:
I forgot to call `Delete`:

https://github.com/dotnet/runtime/blob/ec07ac8709dd5676084a79193cac9498eaa427ff/src/libraries/System.IO.FileSystem/tests/Directory/Delete.cs#L229-L232

Which is overriden in subclasses that test `DirectoryInfo` as well:

https://github.com/dotnet/runtime/blob/57bfe474518ab5b7cfe6bf7424a79ce3af9d6657/src/libraries/System.IO.FileSystem/tests/DirectoryInfo/Delete.cs#L27-L37

 fixed that in https://github.com/dotnet/runtime/pull/56996/commits/bd7f5f0ae71213e39803cec44d0a72ee7699c397

Now having this type partial makes sense.

Reply:
Speaking of using`PlatformSpecific` attribute in files that are conditionally compiled only on certain OSes. I prefer to have the extra attribute, but I don't have a strong opinion about it. @ViktorHofer do we have some guidelines for that?

Reply:
I agree with @adamsitnik. For such situations we usually move the platform specific code into a separate file.